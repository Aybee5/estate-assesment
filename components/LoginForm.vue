<template>
  <div>
    <NuxtLogo class="tw-text-white md:tw-hidden tw-text-logo tw-mb-6" />

    <form @submit.prevent="handleSignIn(email, password)" id="signInForm">
      {{ errorMessage }}
      <div class="field">
        <label class="label">Email</label>
        <p>
          <input
            :class="{ 'tw-border-l-orange-700': hasError }"
            v-model="email"
            type="email"
            placeholder="Enter your email"
            required
          />
        </p>
      </div>
      <div class="field">
        <label for="password" class="label">Password</label>
        <p>
          <input
            :class="{ 'tw-border-l-orange-700': hasError }"
            v-model="password"
            type="password"
            placeholder="Enter your password"
            required
          />
        </p>
      </div>
      <button
        type="submit"
        :class="{ 'tw-border-l-orange-700': isLoading }"
        class="submit-btn tw-mt-5"
      >
        Log in
      </button>
    </form>
  </div>
</template>

<script>
export default {
  data() {
    return {
      email: "",
      password: "",
      hasError: false,
      isLoading: false,
      errorMessage: "",
    };
  },
  methods: {
    async handleSignIn(email, password) {
      this.isLoading = true;
      try {
        await this.$store.dispatch("signIn", {
          email,
          password,
        });
        this.$router.push("/");
      } catch (error) {
        this.hasError = true;
        this.errorMessage = error;
      } finally {
        this.isLoading = false;
      }
    },
  },
};
</script>

<style></style>
