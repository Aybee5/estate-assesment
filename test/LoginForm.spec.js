import { mount } from "@vue/test-utils";
import LoginForm from "@/components/LoginForm.vue";

describe("LoginForm", () => {
  it("renders", () => {
    const wrapper = mount(LoginForm);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
