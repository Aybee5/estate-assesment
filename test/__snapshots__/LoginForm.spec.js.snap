// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders 1`] = `
"<div>
  <nuxtlogo class=\\"tw-text-white md:tw-hidden tw-text-logo tw-mb-6\\"></nuxtlogo>
  <form id=\\"signInForm\\">

    <div class=\\"field\\"><label class=\\"label\\">Email</label>
      <p><input type=\\"email\\" placeholder=\\"Enter your email\\" required=\\"required\\" class=\\"\\"></p>
    </div>
    <div class=\\"field\\"><label for=\\"password\\" class=\\"label\\">Password</label>
      <p><input type=\\"password\\" placeholder=\\"Enter your password\\" required=\\"required\\" class=\\"\\"></p>
    </div> <button type=\\"submit\\" class=\\"submit-btn tw-mt-5\\">
      Log in
    </button>
  </form>
</div>"
`;
